grammar.js

```diff
@@ -329,16 +338,25 @@ module.exports = grammar({
       $._expression
     )),
 
-    list_splat_parameter: $ => seq(
+    _list_splat_expression: $ => prec(PREC.parameter, seq(
       '*',
-      optional($.identifier)
-    ),
+      optional(choice(
+        seq('(',')'),
+        optional($._expression)
+    )))),
 
-    dictionary_splat_parameter: $ => seq(
+    _dictionary_splat_expression: $ => prec(PREC.parameter, seq(
       '*',
       '*',
-      $.identifier
-    ),
+      choice(
+        seq('{','}'),
+        $._expression
+      )
+    )),
+
+    list_splat_parameter: $ => $._list_splat_expression,
+
+    dictionary_splat_parameter: $ => $._dictionary_splat_expression,
```

Comment:
The `list_splat` and `dictionary_splat` changes help us conform to a similar structure for parameters and arguments, and are also examples of matching against `*()` for list and `**{}` for dictionary.